src/components/AppLayout/Drawer/useDrawer.js

```diff
@@ -1,13 +1,13 @@
-import { useState } from "react";
+import { useState, useCallback, useMemo } from "react";
 
 const useDrawer = () => {
   const [shown, setShown] = useState(false);
 
-  const show = () => setShown(true);
+  const show = useCallback(() => setShown(true), []);
 
-  const hide = () => setShown(false);
+  const hide = useCallback(() => setShown(false), []);
 
-  return { shown, show, hide };
+  return useMemo(() => ({ shown, show, hide }), [shown, show, hide]);
```

Comment:
why do we need to use multiple useCallbacks and useMemo here for a boolean type? Optimization comes with a cost.